solver/src/main/java/org/chocosolver/solver/variables/impl/AbstractVariable.java

```diff
@@ -519,7 +522,47 @@ public ICause getCause() {
         return cause;
     }
 
-    static class BipartiteList {
+    static abstract class ABipartiteList {
```

Comment:
Suggestions : 
- externalize those classes to reduce file size
- it seems the abstract class has no field and could then be an interface with default implementation (the default implem could also be done as a separate class).